src/main/java/com/google/devtools/build/lib/analysis/config/FragmentOptions.java

```diff
@@ -100,7 +100,11 @@ protected static List<Map.Entry<String, String>> normalizeEntries(
       List<Map.Entry<String, String>> entries) {
     LinkedHashMap<String, String> normalizedEntries = new LinkedHashMap<>();
     for (Map.Entry<String, String> entry : entries) {
-      normalizedEntries.put(entry.getKey(), entry.getValue());
+      if (entry.getKey() == null) {
+        normalizedEntries.remove(entry.getValue());
```

Comment:
I think all of these places deserve a short one line comment to indicate what's happening (otherwise the fact that you are using the value as the key to remove is not obvious).